app/views/govuk_publishing_components/components/_warning_text.html.erb

```diff
@@ -15,8 +18,15 @@
   <% unless text_icon.empty? %>
     <%= tag.span text_icon, class: "govuk-warning-text__icon", "aria-hidden": "true" %>
   <% end %>
-  <%= tag.strong class: text_classes do %>
-    <%= tag.span text_assistive, class: "govuk-warning-text__assistive" %>
-    <%= text %>
+  <% if heading_level > 0 %>
+    <%= content_tag(shared_helper.get_heading_level, class: text_classes) do %>
+      <%= tag.span text_assistive, class: "govuk-warning-text__assistive" %>
```

Comment:
I wonder if we can combine these two lines into something reusable, as they're always used regardless of whether it's a heading or not?

```
<%= tag.span text_assistive, class: "govuk-warning-text__assistive" %>
<%= text %>
```

Reply:
Yeah, I wondered about that, but everything I tried reduced the two pairs of lines to one but added at least two more lines of code i.e. no actual lines saved. In the end I gave up on the basis that this is at least clear to read. Happy to accept a suggestion though?

Reply:
For me it was less about the number of lines of code and more about having multiple places to update if we want to make changes in the future.

We could maybe pull it out into a helper or just do something like this:
```
<% text_capture = capture do %>
  <%= tag.span text_assistive, class: "govuk-warning-text__assistive" %>
  <%= text %>
<% end %>

...


if heading
    <%= text_capture %>
else
    <%= text_capture %>
```

Reply:
👍